invokeai/app/services/images/images_default.py

```diff
@@ -146,25 +154,22 @@ def get_dto(self, image_name: str) -> ImageDTO:
             self.__invoker.services.logger.error("Problem getting image DTO")
             raise e
 
-    def get_metadata(self, image_name: str) -> ImageMetadata:
+    def get_metadata(self, image_name: str) -> Optional[MetadataField]:
         try:
-            image_record = self.__invoker.services.image_records.get(image_name)
-            metadata = self.__invoker.services.image_records.get_metadata(image_name)
-
-            if not image_record.session_id:
-                return ImageMetadata(metadata=metadata)
-
-            session_raw = self.__invoker.services.graph_execution_manager.get_raw(image_record.session_id)
-            graph = None
-
-            if session_raw:
-                try:
-                    graph = get_metadata_graph_from_raw_session(session_raw)
-                except Exception as e:
-                    self.__invoker.services.logger.warn(f"Failed to parse session graph: {e}")
-                    graph = None
+            return self.__invoker.services.image_records.get_metadata(image_name)
+        except ImageRecordNotFoundException:
+            self.__invoker.services.logger.error("Image record not found")
+            raise
```

Comment:
Do we want to throw instead of return None here?

Reply:
images are allowed to have no metadata so no throw